exercises/reforestation/canonical-data.json

```diff
@@ -0,0 +1,93 @@
+{
+    "exercise": "reforestation",
+    "version": "1.0.0",
+    "cases": [
+      {
+        "description": "Base case: empty tree",
+        "property": "treeTraversals",
+        "input": {
+          "preorder": [],
+          "inorder": []
+        },
+        "expected": {}
+      },
+      {
+        "description": "1 more step than base case",
+        "property": "treeTraversals",
+        "input": {
+            "preorder": ["a"],
+            "inorder": ["a"]
+        },
+        "expected": {"v":"a", "l": {}, "r": {}}
+      },
+      {
+        "description": "Core functionality with realistic size traversal",
+        "property": "treeTraversals",
+        "input": {
+            "preorder": ["a","i","x","f","r"],
+            "inorder": ["i","a","f","x","r"]
+        },
+        "expected": {"v": "a",
+            "l": {"v": "i", "l": {}, "r": {}},
+            "r": {"v": "x",
+                "l": {"v": "f", "l": {}, "r": {}},
+                "r": {"v": "r", "l": {}, "r": {}}
+            }
+        }
+      },
+      {
+        "description": "Traversals must be fully consumed. No extra values.",
+        "property": "treeTraversals",
+        "input": {
+          "preorder": ["a", "b"],
+          "inorder": ["b", "a", "r"]
+        },
+        "expected": {"error": "invalid traversal"}
```

Comment:
Isn't the error that the traversals array are of incorrect length? If so, maybe the error message could reflect this.

Reply:
okay